RequestService/RequestService.Core/Services/ChartDataService.cs

```diff
@@ -0,0 +1,179 @@
+﻿using HelpMyStreet.Contracts.ReportService;
+using HelpMyStreet.Utils.Enums;
+using HelpMyStreet.Utils.Extensions;
+using Newtonsoft.Json;
+using RequestService.Core.Domains;
+using RequestService.Core.Interfaces.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace RequestService.Core.Services
+{
+    public class ChartDataService: IChartDataService
+    {
+        private readonly IRepository _repository;
+
+        public ChartDataService(IRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task<List<DataPoint>> GetActivitiesByMonth(int groupId)
+        {
+            DateTime minDate = DateTime.UtcNow.Date.AddYears(-1);
+
+            var dataItems = await _repository.GetActivitiesByMonth(groupId, minDate);
+
+            //Get distinct activities from list
+            var supportActivities = dataItems.Select(x => x.Series).Distinct().ToList();
+
+            List<DataPoint> dataPoints = new List<DataPoint>();
+
+            //Populate chartitems with support activities for each month
+            while (minDate < DateTime.UtcNow.Date)
```

Comment:
Is it deliberate to exclude the current month?

Reply:
Actually, maybe we're only excluding the current month when it's the first of the month?  Hopefully there's a nice unit test to validate the behaviour 😉